default.nix

```diff
@@ -19,14 +20,10 @@ let gitignoreSrc = import (pkgs.fetchFromGitHub {
     }) {};
     nix-thunk = import ./deps/nix-thunk {};
     chainwebDataSrc = pkgs.lib.cleanSourceWith {
-      filter = path: type:
-        # Filter out these files so that nix doesn't recompile CW-D every time
-        # these files change, they're not relevant to the Haskell build anyway.
-        let ignored = map toString [./docker.nix ./default.nix];
-        in !(builtins.elem path ignored);
-      src = gitignoreSrc.gitignoreSource ./.;
+      src = gitignoreSrc.gitignoreSource ./haskell-src;
```

Comment:
Nice, seems like `gitignoreSource` can deal with `.gitignore` files outside the folder you pass into it, so this works:
```
$ nix-build 
/nix/store/2xrchp7xyjiyjpk1rq72cijd381jii57-chainweb-data-2.1.1
$ touch haskell-src/phony.dump-hi
$ nix-build 
/nix/store/2xrchp7xyjiyjpk1rq72cijd381jii57-chainweb-data-2.1.1
enis@enis-dt-ubuntu:~/local/chainweb-data$ 
```